### PR TITLE
[Haxe][CodeGenerator] Added new feature `Convert static call into using call`. fixes #1565

### LINKED ISSUE #1565: [Haxe] Converting static method call into static extension call with codegenerator
SomeClass.some|Method(some);   where | is cursor position
press ctrl + shift + 1 -> convert to using
enter
should convert 
SomeClass.someMethod(some) 
into
some.someMethod();

### PR DESCRIPTION
**Pull request by SlavaRa**

### CODE REVIEW

**Comment by Neverbirth** [comment]
Conflicts now. I will try to play around a bit with this tomorrow.

**Comment by SlavaRa** [comment]
Fixed

File: External/Plugins/ASCompletion/Completion/ASGenerator.cs [hunk 1]
```diff
@@ -1,619 +1,622 @@
...
+                    {
+                        Match m = Regex.Match(text, String.Format(patternMethodDecl, contextToken));
+                        Match m2 = Regex.Match(text, String.Format(patternMethod, contextToken));
+                        if (!m.Success && m2.Success)
+                        {
+                            contextMatch = m;
+                            ShowChangeMethodDeclList(found, options);
+                        }
+                    }
+                    else if (resolve != null
```
Comment (Neverbirth) [inline External/Plugins/ASCompletion/Completion/ASGenerator.cs]: These new checks could go directly to the parent else.

File: PluginCore/PluginCore/Resources/ja_JP.resx [hunk 2]
```diff
@@ -6288,4 +6288,8 @@ AND - All conditions are met.</value>
     <value>Sets the maximum number of processes for diagnostics (linting) that run at the same time. Defaults to 1.</value>
     <comment>Added after 5.2.1</comment>
   </data>
+  <data name="HaXeContext.Label.ConvertStaticCallToUsingCall" xml:space="preserve">
+    <value>Convert to usgin</value>
```
Comment (Neverbirth) [inline PluginCore/PluginCore/Resources/ja_JP.resx]: Misspelled in all the languages.

Comment (SlavaRa) [reply PluginCore/PluginCore/Resources/ja_JP.resx]: ?

**Comment by SlavaRa** [comment]
I think this can be released in a separate plugin :)

**Pull request closed without merging**